lib/roby/plan.rb

```diff
@@ -1609,6 +1609,15 @@ def finalize_event(event, timestamp = nil)
             event.finalized!(timestamp)
         end
 
+        # Apply to +to+ the marks (permanent, mission) of +from+
+        #
+        # It does not remove any marks from +from+
+        def copy_marks(to, from)
```

Comment:
Alright ... my deepest apologies.

I think your original hunch was right, and using keyword arguments is strictly better ... Removes a big source of potential errors.

Just use `from` and `to` and call the methods `copy_task_marks` though. More aligned to e.g. `add_permanent_task` or `unmark_permanent_task`.